server/methods.js

```diff
@@ -1,91 +1,174 @@
-            userNick: getNickname(Meteor.user()),
-            userAvatar: getAvatar(Meteor.user())
-        }).save();
-    },
 
-    "userroom.build" (roomName) {
-        const userId = Meteor.userId();
-        const roomId = Chatter.Room.findOne({"name": roomName})._id;
-        const records = Chatter.UserRoom.find({"userId": userId, "roomId" : roomId}).fetch();
-
-        if (records === undefined || records.length === 0) {
-            return new Chatter.UserRoom({
-                roomId: roomId,
-                userId: userId
-            }).save();
-        }
-    },
+  "user.check" () {
+    const chatterUsers = Chatter.User.find({userId: Meteor.userId()}).fetch();
+    return chatterUsers.length > 0;
+  },
 
-    "room.build" (form) {
-        const user = Meteor.user();
-        return new Chatter.Room({
-            name: form.name,
-            roomType: form.roomType
-        }).save();
-    },
+  "message.build" (params) {
+    check(params, {
+      message: String,
+      roomId: String
+    });
 
-    "room.update" (id) {
-        Chatter.Room.update({
-            _id : id
-        },
-        {
-            $set:{lastActive : new Date()}
-        });
-    },
+    const chatterId = getChatterId(Meteor.userId());
 
-    "room.users" (roomId) {
-        const userRooms = Chatter.UserRoom.find({"roomId": roomId}).fetch();
-        const users = userRooms.map(function(userRoom) {
-            const user = Meteor.users.findOne({_id: userRoom.userId });
-            const userInfo = {
-                _id: user._id,
-                userNick: getNickname(user)
-            };
-            return userInfo;
-        });
-        return users;
-    },
+    if (!userInRoom(chatterId, params.roomId)) {
+      throw new Meteor.Error("user-not-in-room", "user must be in room to post messages");
+    }
 
-    "userroom.count.reset" (userId, roomId) {
-        return Chatter.UserRoom.update({
-            roomId : roomId,
-            userId: userId
-        },
-        {
-            $set:{count: 0}
-        });
-    },
+    const message = new Chatter.Message({
+      message: params.message,
+      roomId: params.roomId,
+      userId: chatterId
+    });
 
-    "userroom.count.increase" (userId, roomId) {
-        const userRooms = Chatter.UserRoom.find({"roomId": roomId, "userId": {$nin:[userId]}}).fetch();
-        const users = userRooms.map(function(userRoom) {
-            Chatter.UserRoom.update({
-                roomId : userRoom.roomId,
-                userId: userRoom.userId
-            },
-            {
-                $inc: { count: 1}
-            });
-        });
+    if (message.validate()) {
+      return message.save();
     }
-});
 
-function getNickname(user) {
-    const nickPath = Chatter.options.nickProperty;
-    const nick = nickPath.split('.').reduce((prevVal, el) => prevVal[el] , user);
-    return nick;
-}
+    message.throwValidationException();
+  },
+
+  "userroom.build" (form) {
```

Comment:
noted and changed, decided to use names like room.join even if the changes are still being done to the UserRoom model